ml-proto/src/lexer.mll

```diff
@@ -210,35 +210,34 @@ rule token = parse
   | "gt."(fxx as t) { COMPARE (floatop t F32.Gt F64.Gt) }
   | "ge."(fxx as t) { COMPARE (floatop t F32.Ge F64.Ge) }
 
-  | "converts."(ixx as t)".i32" { CONVERT (intop t I32.ToInt32S I64.ToInt32S) }
-  | "convertu."(ixx as t)".i32" { CONVERT (intop t I32.ToInt32U I64.ToInt32U) }
-  | "converts."(ixx as t)".i64" { CONVERT (intop t I32.ToInt64S I64.ToInt64S) }
-  | "convertu."(ixx as t)".i64" { CONVERT (intop t I32.ToInt64U I64.ToInt64U) }
-  | "converts."(ixx as t)".f32"
-    { CONVERT (intop t I32.ToFloat32S I64.ToFloat32S) }
-  | "convertu."(ixx as t)".f32"
-    { CONVERT (intop t I32.ToFloat32U I64.ToFloat32U) }
-  | "converts."(ixx as t)".f64"
-    { CONVERT (intop t I32.ToFloat64S I64.ToFloat64S) }
-  | "convertu."(ixx as t)".f64"
-    { CONVERT (intop t I32.ToFloat64U I64.ToFloat64U) }
-  | "converts."(fxx as t)".i32"
-    { CONVERT (floatop t F32.ToInt32S F64.ToInt32S) }
-  | "convertu."(fxx as t)".i32"
-    { CONVERT (floatop t F32.ToInt32U F64.ToInt32U) }
-  | "converts."(fxx as t)".i64"
-    { CONVERT (floatop t F32.ToInt64S F64.ToInt64S) }
-  | "convertu."(fxx as t)".i64"
-    { CONVERT (floatop t F32.ToInt64U F64.ToInt64U) }
-  | "convert."(fxx as t)".f32"
-    { CONVERT (floatop t F32.ToFloat32 F64.ToFloat32) }
-  | "convert."(fxx as t)".f64"
-    { CONVERT (floatop t F32.ToFloat64 F64.ToFloat64) }
-
-  | "cast.i32.f32" { CONVERT (Values.Int32 I32.ToFloatCast) }
-  | "cast.i64.f64" { CONVERT (Values.Int64 I64.ToFloatCast) }
-  | "cast.f32.i32" { CONVERT (Values.Float32 F32.ToIntCast) }
-  | "cast.f64.i64" { CONVERT (Values.Float64 F64.ToIntCast) }
+  | "extends.i32.i64" { CONVERT (Values.Int64 I64.ExtendSInt32) }
```

Comment:
While you said you're waiting to move result type to the front, perhaps you could switch `*s` to `_signed` in this PR?